File: app/src/test/java/pl/srw/billcalculator/settings/restore/ConfirmRestoreSettingsPresenterTest.java
package pl.srw.billcalculator.settings.restore;

import org.junit.Before;
import org.junit.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import pl.srw.billcalculator.settings.prices.RestorablePrices;
import pl.srw.billcalculator.util.ProviderMapper;

import static org.mockito.Mockito.verify;

public class ConfirmRestoreSettingsPresenterTest {
    @InjectMocks private ConfirmRestoreSettingsPresenter sut;
    @Mock private RestorablePrices prices;
    @Mock private ConfirmRestoreSettingsPresenter.ConfirmRestoreSettingsView view;
    @Mock private ProviderMapper providerMapper;

    @Before
    public void setUp() throws Exception {
        sut = new ConfirmRestoreSettingsPresenter(providerMapper);
        MockitoAnnotations.initMocks(this);
    }

    @Test
    public void onConfirmedClicked_setDefaultPrices() throws Exception {
        // WHEN
        sut.onConfirmedClicked();

        // THEN
        verify(prices).setDefault();
    }

    @Test
    public void onConfirmedClicked_refreshView() throws Exception {
        // WHEN
        sut.onConfirmedClicked();

        // THEN
        verify(view).refresh();
    }
}

File: app/src/test/java/pl/srw/billcalculator/settings/details/restore/ConfirmRestoreSettingsPresenterTest.kt
package pl.srw.billcalculator.settings.details.restore

import com.nhaarman.mockito_kotlin.mock
import com.nhaarman.mockito_kotlin.verify
import com.nhaarman.mockito_kotlin.verifyZeroInteractions
import org.junit.Before
import org.junit.Test
import pl.srw.billcalculator.data.settings.prices.PricesRepo
import pl.srw.billcalculator.type.Provider

class ConfirmRestoreSettingsPresenterTest {

    val provider = Provider.PGNIG
    val pricesRepo: PricesRepo = mock()
    val sut = ConfirmRestoreSettingsPresenter(pricesRepo)

    @Before
    fun setUp() {
        sut.setup(provider)
    }

    @Test
    fun `set default prices on confirm restore`() {
        sut.onConfirmClicked()

        verify(pricesRepo).setDefaultPricesFor(provider)
    }

    @Test
    fun `don't change prices on restore canceled`() {
        sut.onCancelClicked()

        verifyZeroInteractions(pricesRepo)
    }
}
